cli/flox/src/commands/init/python.rs

```diff
@@ -322,7 +320,6 @@ impl PoetryPyProject {
             .ok_or_else(|| {
                 anyhow!("No python version specified at 'tool.poetry.dependencies.python'")
             })?
-            .parse::<semver::VersionReq>()?
```

Comment:
(comment, nb) It seem odd to me that we do not try to normalize inputs as early as possible if we can.
I get that we want to avoid parsing the version twice but id also lean for a stricter manifest format.

Reply:
The manifest can't be stricter because there isn't a complete overlap with what's supported by the Rust library in the CLI and the Python library in the catalog (after https://github.com/flox/catalog-server/issues/264) and we have more than one source of the information (manifest and init hooks).

Reply:
> yeah you'd think semver is specified enough for there to be compatible impls 😑

Standards, ha!

> I'm not sure how [flox/catalog-server#264](https://github.com/flox/catalog-server/issues/264) relates to this, it seems to do the same normalization server side.. why do we need this on both ends?

It's not the same normalization because there isn't overlap in the spreadsheet, e.g.

- CLI can't parse Python formats of `==3.11` or `~=3.11` when the catalog does via `SimpleSpec`
- CLI can't parse NodeJS formats of `>=3.11 <3.12` or `3.11 - 3.12` when the catalog could via `NpmSpec`